src/collider/Geom.cpp

```diff
@@ -73,160 +68,161 @@ void Geom::CollideSphere(Sphere &sphere, void (*callback)(CollisionContact *)) c
-
-			vector3d _dir(
-				double(edges[edgeNode->triIndicesStart[i]].dir.x),
-				double(edges[edgeNode->triIndicesStart[i]].dir.y),
-				double(edges[edgeNode->triIndicesStart[i]].dir.z));
-			_dir = transToB.ApplyRotationOnly(_dir);
-			dir = vector3f(&_dir.x);
-			isect.dist = edges[edgeNode->triIndicesStart[i]].len;
-			isect.triIdx = -1;
-
-			b->GetGeomTree()->TraceRay(btriNode, _from, dir, &isect);
-
-			if (isect.triIdx == -1) continue;
-			numContacts++;
-			const double depth = edges[edgeNode->triIndicesStart[i]].len - isect.dist;
-			// in world coords
-			CollisionContact contact;
-			contact.pos = b->GetTransform() * (v1 + vector3d(&dir.x) * double(isect.dist));
-			vector3f n = b->m_geomtree->GetTriNormal(isect.triIdx);
-			contact.normal = vector3d(n.x, n.y, n.z);
-			contact.normal = b->GetTransform().ApplyRotationOnly(contact.normal);
-			contact.distance = isect.dist;
-
-			contact.depth = depth;
-			contact.triIdx = isect.triIdx;
-			contact.userData1 = m_data;
-			contact.userData2 = b->m_data;
-			// contact geomFlag is bitwise OR of triangle's and edge's flags
-			contact.geomFlag = b->m_geomtree->GetTriFlag(isect.triIdx) |
-				edges[edgeNode->triIndicesStart[i]].triFlag;
-			callback(&contact);
-			if (--maxContacts <= 0) return;
-		}
-	} else {
-		CollideEdgesTris(maxContacts, edgeNode->kids[0], transToB, b, btriNode, callback);
-		CollideEdgesTris(maxContacts, edgeNode->kids[1], transToB, b, btriNode, callback);
+	const GeomTree::Edge *edges = GetGeomTree()->GetEdges();
+	const std::vector<vector3f> &rVertices = GetGeomTree()->GetVertices();
+
+	const GeomTree::Edge &edge = edges[edgeIdx];
+	const vector3d v1 = transToB * vector3d(rVertices[edge.v1i]);
+	const vector3f _from(float(v1.x), float(v1.y), float(v1.z));
+
+	vector3d dir = transToB.ApplyRotationOnly(vector3d(edge.dir));
+	isect_t isect;
+
+	isect.dist = edge.len;
+	isect.triIdx = -1;
+
+	isect_buf.clear();
+	b->GetGeomTree()->GetTriTree()->TraceRay(v1, 1.0 / dir, edge.len, isect_buf, triNode);
+
+	// TODO
+	// This is subtly dependent on overwriting intersections with the last triangle to be processed
+	// Ideally all contacts here would be reported...
+	vector3f traceDir = vector3f(dir);
+	for (uint32_t tri : isect_buf) {
+		b->GetGeomTree()->RayTriIntersect(1, _from, &traceDir, tri * 3, &isect);
 	}
+
+	if (isect.triIdx == -1)
+		return;
+
+	// in world coords
+	CollisionContact contact;
+	contact.pos = b->GetTransform() * (v1 + dir * double(isect.dist));
+	contact.normal = vector3d(b->m_geomtree->GetTriNormal(isect.triIdx));
+	contact.normal = b->GetTransform().ApplyRotationOnly(contact.normal);
+	contact.distance = isect.dist;
+
+	contact.depth = edge.len - isect.dist;
+	contact.triIdx = isect.triIdx;
+	contact.userData1 = m_data;
+	contact.userData2 = b->m_data;
+	// contact geomFlag is bitwise OR of triangle's and edge's flags
+	contact.geomFlag = b->m_geomtree->GetTriFlag(isect.triIdx) | edge.triFlag;
+
+	contacts.push_back(contact);
```

Comment:
`contact.timestep` is not defined.

Reply:
Existing behavior. Resolved with explicit zero-initialization.